sparksql-scalapb/src/main/scala/scalapb/spark/CatalystMapper.scala

```diff
@@ -0,0 +1,80 @@
+package scalapb.spark
+
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.expressions.objects.StaticInvoke
+import org.apache.spark.sql.types.{DataType, ObjectType, TimestampType}
+import scalapb.GeneratedMessageCompanion
+import scalapb.descriptors.{Descriptor, FieldDescriptor, PInt, PLong, PMessage, PValue}
+
+import scala.collection.immutable
+
+trait CatalystMapper {
+  def convertedType(descriptor: Descriptor): Option[DataType]
+  def fromCatalyst(
+      cmp: GeneratedMessageCompanion[_],
+      input: Expression,
+      helpers: FromCatalystHelpers
+  ): immutable.Seq[Expression]
```

Comment:
Why is the return type `Seq[Expression]` and not just `Expression` ?

Reply:
It's because: https://github.com/ZipRecruiter/sparksql-scalapb/blob/1705cec4a846cf4a7b64e892a36a9c4cd3004641/sparksql-scalapb/src/main/scala/scalapb/spark/FromCatalystHelpers.scala#L30. I gonna do some refactoring on this method to pull out the common code to construct a PMessage.

Reply:
Would love to see if this method can be simplified if we use Injectors.

Reply:
Looks like we can't. I wrote it with `Injectors` and it fails with `ClassCastExceptions`, and the way to fix those exceptions are to wrap the generated `TypeEncoder` with some catalyst expressions. Given the extra work, I think it's better to stick with the more general version of `TypeEncoder`.